### ISSUE TITLE
Use the modern save file dialog

### ISSUE THREAD

**Issue by ExE-Boss** [Franiac/TwitchLeecher#145]
As said [here](https://www.reddit.com/r/Windows10/comments/6b3ntc/), the currently used select download folder dialog is long past its expiration date.

Also, it is impossible to paste the full folder path into the folder text box, meaning that you are forced to use this outdated dialog to find the folder you are looking for every single time you want to download something from Twitch.

Comparison of the dialogs:
================

The outdated dialog:
---------------------
![The outdated “Select Folder” dialog (picture is from the Reddit link)](https://i.redd.it/3n7n34z9mgxy.png "The outdated “Select Folder” dialog (picture is from the Reddit link)")

### Pros:
- Nothing

### Cons:
- You can’t `Ctrl + V` or type in the path, instead, you have to painstakingly navigate there.
- The window is small and can’t be resized.

The modern dialog:
-------------------
![The modern “Select Folder” dialog as used by TortoiseGit](https://cloud.githubusercontent.com/assets/3889017/26735924/4f586ff8-47c4-11e7-8e5e-e82d22468829.png "The modern “Select Folder” dialog as used by TortoiseGit")
[TortoiseGit](https://tortoisegit.org/ "TortoiseGit – Windows Shell Interface to Git") uses the modern dialog.

### Pros:
- You can `Ctrl + V` or type in the path.
- The window can be resized to any arbitrary size.
- Fully compatible with [Aero Glass](http://glass8.eu/).

### Cons:
- Nothing.

**Comment by Franiac** [comment]
TL is a software piece developed by me to fix a bug in my streaming software (was not able to save streams locally while streaming). That's the history of it and you will still find some "fuck it but it works"-parts in it. It was not planned to be released to the public.

TL is implemented if .NET/WPF. WPF itself does not provide an OpenFolderDialog. That's why I use the Windows Forms dialog (that redirects to the old native one) from the good old days. The given dialog you suggest is a native dialog that has to be integrated into TL first because it is not part of .NET or in other words there's no wrapper for it (the reason why I didn't do it in the first place because I wanted to get stuff done quick). I needed a folder dialog and the old one did it for me.

Why would you select a dowhnload folder for every VOD you download? The preferences provide a default download folder that is used by default for every download. Usually you set it once and never touch the download folder again. That's at least how 99% of the users do it. Maybe you missed this option?

I agree that the new dialog is better in terms of navigation and size. But you have to admit that "piece of shit" and "needs desperate replacement" is a little bit off the scale. Like I said, normally you use the dialog once in TL and that's it.

To sum this up: I have limited free time to spend on TL right now (building a house) and I definitely feel the pain but TL has bigger issues that have to be addressed first. But feel free to contribute.
